lib/gauge_fix_fft.cu

```diff
@@ -185,14 +185,26 @@ namespace quda {
   void gaugeFixingFFT(GaugeField& data, int Nsteps, int verbose_interval,
                       double alpha0, int autotune, double tolerance, int stopWtheta)
   {
+    // We hardcode the value of autotune to 1 at this point as it is the last
+    // point before computation begins. This ensures that the the user cannot
+    // override alpha autotuning. This is done because it is very easy for the
+    // FFT gauge fixing to fail with a poorly chosen value of alpha, but
+    // autotuning ensures optimal behaviour.
+    // Users who wish to change this behaviour may remove the follwing line
+    // of code and recompile to regain control of alpha autotuning.
+    autotune = 1;
```

Comment:
What is the best approach with regards to the interface function signatures for gauge fixing? If we go ahead and change them, we may as well consolidate the two functions into a single call to `computeQudaGaugeFixing`, add a `QUDA_GAUGEFIX_TYPE` and direct the workflow within it. We could preserve the two existing function signatures too as wrappers to this consolidated function for back compatibility, or force downstream modifications.

Reply:
As far as I can tell, the MILC interface would be largely unaffected as we simply change the inner plumbings of
```C++
void qudaGaugeFixingOVR(...)
```
and
```C++
void qudaGaugeFixingFFT(...)
```

Reply:
Having a single gauge fixing interface function sounds like a great idea.  Agree regarding MILC itself not needing to be changed, since it goes through milc_interface.cpp, so we can keep these changes internal to QUDA.

Since this breaking change will align with QUDA 2.0, e.g., a major version number change, I think there's no need to maintain the prior legacy interface as a fallback.  So I think we can just remove the prior interface.

Reply:
https://github.com/lattice/quda/commit/aed7763076cef871a73157f6c358e2a053ec6c83